cachi2/core/models/sbom.py

```diff
@@ -184,6 +185,12 @@ def libs_to_packages(libraries: list[Component]) -> list[SPDXPackage]:
             args = dict(annotator=annottr, annotationDate=now, annotationType="OTHER")
             pAnnotation = partial(SPDXPackageAnnotation, **args)
 
+            def get_spdxid(component: Component) -> str:
+                spdxid = f"SPDXRef-Package-{component.name}-{component.version}-{package_hash}"
+                # According to a validation error from https://tools.spdx.org/app/validate/,
+                # the allowed format for SPDXID is SPDXRef-[0-9a-zA-Z\.\-\+]+
+                return re.sub(r"[^0-9a-zA-Z\.\-\+]", "-", spdxid)
```

Comment:
nitpick: I think the interface would be more ergonomic and more re-usable (do we need that?) if it were a blackbox accepting strings directly and only performed the transformation in accordance to the spec. It would require a minor change in commit 2/2, but that's a trivial change anyway.

Reply:
Refactored as suggested